Source/WebCore/platform/graphics/filters/FilterTargetSwitcher.h

```diff
@@ -47,10 +47,10 @@ class FilterTargetSwitcher {
     virtual bool needsRedrawSourceImage() const { return false; }
 
     virtual void beginClipAndDrawSourceImage(GraphicsContext& destinationContext, const FloatRect&) { beginDrawSourceImage(destinationContext); }
-    virtual void endClipAndDrawSourceImage(GraphicsContext& destinationContext) { endDrawSourceImage(destinationContext); }
+    virtual void endClipAndDrawSourceImage(GraphicsContext& destinationContext, const FilterResultsProvider& resultsProvider = { }) { endDrawSourceImage(destinationContext, resultsProvider); }
```

Comment:
I'm confused by these default argument functions, which can't return a `FilterResults&`, so what happens if they get called?

Reply:
The `FilterResultsProvider` will be passed down till it reaches `GraphicsContext::drawFilteredImageBuffer()` where `FilterResults` is passed to `Filter::apply()`:
```
    FilterResults results;
    auto result = filter.apply(sourceImage, sourceImageRect, resultsProvider ? resultsProvider() : results);
```
If the default `FilterResultsProvider` is passed this will be equivalent to:
```
    FilterResults results;
    auto result = filter.apply(sourceImage, sourceImageRect, results);
```
This means the `FilterResults` will be discarded once the resulted `ImageBuffer` is generated and drawn to `GraphicsContext`.